e2e/support/helpers/e2e-notebook-helpers.ts

```diff
@@ -0,0 +1,201 @@
+import { popover } from "e2e/support/helpers/e2e-ui-elements-helpers";
+
+/**
+ * Switch to a notebook editor from a simple query view (aka "chill mode").
+ */
+export function openNotebook() {
+  return cy.findByTestId("qb-header-action-panel").icon("notebook").click();
+}
+
+/**
+ * Select a specific notebook step like filter, join, breakout, etc.
+ *
+ * @see {@link https://github.com/metabase/metabase/pull/17708#discussion_r700082403}
+ */
+export function getNotebookStep(
+  type:
+    | "summarize"
+    | "join"
+    | "expression"
+    | "filter"
+    | "sort"
+    | "limit"
+    | "data",
+  { stage = 0, index = 0 } = {},
+): Cypress.Chainable<JQuery<HTMLElement>> {
+  return cy.findByTestId(`step-${type}-${stage}-${index}`);
+}
+
+/**
+ * @summary Visualize notebook query results.
+ *
+ * This helper intelligently waits for the query to load, and gives you an option
+ * to assert on the waited query response.
+ *
+ * @example
+ * visualize();
+ *
+ * visualize(response => {
+ *   expect(response.body.error).to.not.exist;
+ * });
+ */
+export function visualize(callback?: (response?: object) => void) {
+  cy.intercept("POST", "/api/dataset").as("dataset");
+
+  cy.button("Visualize").click();
+
+  cy.wait("@dataset").then(({ response }) => {
+    if (callback) {
+      callback(response);
+    }
+  });
+}
+
+/**
+ * Summarize (Aggregate).
+ *
+ * Doesn't support summarizing using Custom Expression!
+ */
+export function addSummaryField({
+  metric,
+  table,
+  field,
+  stage = 0,
+  index = 0,
+}: {
+  metric:
+    | "Count of rows"
+    | "Sum of ..."
+    | "Average of ..."
+    | "Number of distinct values of ..."
+    | "Cumulative sum of ..."
+    | "Cumulative count of rows"
+    | "Standard deviation of ..."
+    | "Minimum of ..."
+    | "Maximum of ...";
+  table?: string;
+  field?: string;
+  stage?: number;
+  index?: number;
+}) {
+  getNotebookStep("summarize", { stage, index })
+    .findByTestId("aggregate-step")
+    .findAllByTestId("notebook-cell-item")
+    .last()
+    .click();
+
+  popover().within(() => {
+    cy.findByText(metric).click();
+    if (table) {
+      cy.findByText(table).click();
+    }
+    if (field) {
+      cy.findByText(field).click();
+    }
+  });
+}
+
+/**
+ * Group by.
```

Comment:
It's called a [breakout](https://github.com/metabase/metabase/wiki/Metabase-Glossary).

Same for `removeSummaryGroupingField`.